### ISSUE TITLE
GNAT Invalid command line option, can't compile Ada code

### ISSUE THREAD

**Issue by lesscodeless** [NixOS/nixpkgs#30533]
## Issue description
Compiling a trivial Ada application fails due to invalid command line options added by some nix-specific wrapper scripts.

In other Linux distributions 'gnatmake' is a binary file, in nix however it`s a wrapper script.

### Steps to reproduce
```
$ gnatmake hello.adb 
gcc -c -I/nix/store/2dam02rdvc2bp5ghcd7x8ivrd2dwsl3p-gnat-4.5.4/lib/gcc/x86_64-unknown-linux-gnu/4.5.4/adainclude hello.adb
/nix/store/w7n604zqlncab02gxyf5kld19wl5v8k6-gnat-wrapper-4.5.4/bin/gnatbind -aI/nix/store/2dam02rdvc2bp5ghcd7x8ivrd2dwsl3p-gnat-4.5.4/lib/gcc/x86_64-unknown-linux-gnu/4.5.4/adainclude -aO/nix/store/2dam02rdvc2bp5ghcd7x8ivrd2dwsl3p-gnat-4.5.4/lib/gcc/x86_64-unknown-linux-gnu/4.5.4/adalib -x hello.ali
/nix/store/w7n604zqlncab02gxyf5kld19wl5v8k6-gnat-wrapper-4.5.4/bin/gnatlink hello.ali
gnat1: warning: command line option "-Wformat" is valid for C/C++/ObjC/ObjC++ but not for Ada
gnat1: warning: command line option "-Wformat-security" is valid for C/C++/ObjC/ObjC++ but not for Ada
gnat1: error: unrecognized command line option "-fstack-protector-strong"
gnatmake: *** link failed.
```

where 'hello.adb' is a simple "hello world" program with the following contents:

```
with Ada.Text_IO;

procedure Hello is
begin
  Ada.Text_IO.Put_Line("Hello");
end;
```

## Technical details

* System:
```
$ nixos-version 
17.09.1756.c99239bca0 (Hummingbird)
```

* Nix version:
```
$ nix-env --version
nix-env (Nix) 1.11.15
```

* Nixpkgs version:
```
$ nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"17.09.1756.c99239bca0"
```

* Sandboxing enabled:
```
$ grep build-use-sandbox /etc/nix/nix.conf
build-use-sandbox = false
```

**Comment by lesscodeless** [comment]
**Additional information**
As can be seen in the trail below it's the hardning script that adds the invalid command line options.

/nix/store/w7n604zqlncab02gxyf5kld19wl5v8k6-gnat-wrapper-4.5.4/bin/gnatlink
`extraAfter=("--GCC=/nix/store/w7n604zqlncab02gxyf5kld19wl5v8k6-gnat-wrapper-4.5.4/bin/gcc")`

/nix/store/w7n604zqlncab02gxyf5kld19wl5v8k6-gnat-wrapper-4.5.4/bin/gnatlink
`source /nix/store/w7n604zqlncab02gxyf5kld19wl5v8k6-gnat-wrapper-4.5.4/nix-support/add-hardening.sh`

/nix/store/w7n604zqlncab02gxyf5kld19wl5v8k6-gnat-wrapper-4.5.4/nix-support/add-hardening.sh
`hardeningFlags=(fortify stackprotector pic strictoverflow format relro bindnow)`

**Comment by bjornfor** [comment]
Refs. https://github.com/NixOS/nixpkgs/issues/18995, https://github.com/NixOS/nixpkgs/pull/28029.

**Comment by vcunat** [comment]
TL;DR: work around via `export hardeningDisable=all`

**Comment by vcunat** [comment]
Our default gnat is in version 9 now, so these errors shouldn't happen.  To be sure, `gnatmake hello.adb` works for me.
